File: 3277-find-the-number-of-ways-to-place-people-ii/find-the-number-of-ways-to-place-people-ii.py
class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        n = len(points)
        points.sort(key = lambda point: (point[0], -point[1]))

        result = 0
        for i in range(n):
            y = float("-inf")
            for j in range(i + 1, n):
                if points[i][1] >= points[j][1] and points[j][1] > y:
                    result += 1
                    y = points[j][1]

        return result

File: 3277-find-the-number-of-ways-to-place-people-ii/find-the-number-of-ways-to-place-people-ii.cpp
class Solution {
public:
    int numberOfPairs(vector<vector<int>>& points) {
        int n = points.size();
        sort(points.begin(), points.end(), [](const auto& a, const auto& b) {
            if (a[0] != b[0]) {
                return a[0] < b[0];
            } else {
                return a[1] > b[1];
            }
        });

        int result = 0;
        for (int i = 0; i < n; i++) {
            int y = INT_MIN;
            for (int j = i + 1; j < n; j++) {
                if (points[i][1] >= points[j][1] && points[j][1] > y) {
                    result++;
                    y = points[j][1];
                }
            }
        }

        return result;
    }
};